docs/weekly-questions/joes-water-corporation.md

```diff
@@ -117,129 +117,99 @@ There is only one store, where we have a higher price than our competitor. Thus
 When we apply the discount, our price decreases to $5$. However, as our competitor wins in the case of a tie, we still make $0$ profit.
 Therefore, as our profit did not change, the output is `NO`.
 
-<!--
-## Model Solution
+
+## Model Solution 
 
 <details><summary>Click to reveal</summary>
 <p>
 
 ```py
-def compute_total_profit(prices, competitors_prices):
-    return sum(
-        price
-        for price, competitors_price in zip(prices, competitors_prices)
-        if price < competitors_price
-    )
-
-prices = list(map(int, input().split()))
-competitors_prices = list(map(int, input().split()))
+joes_prices = input().split(' ')
+competitions_prices = input().split(' ')
 discount = int(input())
 
-original_profit = compute_total_profit(prices, competitors_prices)
-prices = [price - discount for price in prices]
-after_discount_applied = compute_total_profit(prices, competitors_prices)
+original_profit = 0
+discounted_profit = 0
 
-print("YES" if after_discount_applied > original_profit else "NO")
-```
+for i in range(len(joes_prices)):
+    cur_joe_price = int(joes_prices[i])
+    cur_competitor_price = int(competitions_prices[i])
+    # These conditionals check for 
+    if cur_joe_price < cur_competitor_price:
+        original_profit += cur_joe_price
+    if cur_joe_price - discount < cur_competitor_price:
+        discounted_profit += cur_joe_price - discount
 
+print('YES' if discounted_profit > original_profit else 'NO')
+```
 ### Step-by-step tutorial
 
 Let's begin by reading in the input.
-
 ```py
-prices = map(int, input().split())
-competitors_prices = map(int, input().split())
+joes_prices = input().split(' ')
+competitions_prices = input().split(' ')
```

Comment:
The `map()` construct is cleaner and we've already introduced it; please use that instead.